virtcontainers/kata_agent.go

```diff
@@ -1252,8 +1252,15 @@ func (k *kataAgent) resumeContainer(sandbox *Sandbox, c Container) error {
 }
 
 func (k *kataAgent) onlineCPUMem(cpus uint32, cpuOnly bool) error {
+	wait := true
+	k.Logger().WithFields(
+		logrus.Fields{
+			"cpus":    cpus,
+			"cpuOnly": cpuOnly,
+			"Wait":    wait,
+		}).Debugf("request to  agent to online cpus and memory")
 	req := &grpc.OnlineCPUMemRequest{
-		Wait:    false,
+		Wait:    wait,
```

Comment:
boot time could be slower with this change, have you measured `docker run --cpus=8 -ti busybox true` with `Wait: true` and `Wait: false`?

Reply:
I'll throw  in the obligatory reference to https://github.com/kata-containers/tests/tree/master/metrics/report where you can use the grabdata (and tell it to only measure boot time if you want) to do a before/after comparison.... ;-)

Reply:
@devimc I suspect this could happend while we increase the amount of vcpus hotplugged to the VM.

So I was wondering if cpuset should be deferred as well. Let me try with server with more cpus

Reply:
Any results regarding this @jcvenegas?
Also, what's the rationale behind this change?

Reply:
@jcvenegas the problem with `Wait: true` is that the agent waits for the CPUs and the runtime has to wait for the agent, IMO resources should be connected in background, in the case of CPUs I haven't seen any problem with this behaviour, but I guess you detected something with the memory.

Reply:
@jcvenegas we should not keep some racy code. So I would suggest that you use `wait = true` but run this in a goroutine that you can synchronize with at some point. And by that, I mean that we should allow parallel stuff to happen while the CPU gets onlined, but have to wait before to update cpuset.
Please open a separate issue and PR for that.

In the meantime, and because this bug is not introduced by this PR, I don't see why we could not merge it.